From 67b499d637014365fb13621285db88e89b8ddafb Mon Sep 17 00:00:00 2001
From: Jeremy Hitt <jeremy.hitt@isilon.com>
Date: Fri, 13 Feb 2026 00:58:14 +0000
Subject: [PATCH] Fix null dereference in SPNEGO token processing

Commit fdceb225f881e2b1337eebcb9a9443fa4a9be3fd erroneously removed a
check from get_negTokenResp() for successful decoding of the
mechListMIC field.  Restore the check to prevent a null pointer
dereference.

[ghudson@mit.edu: rewrote commit message]

(cherry picked from commit 4ae75cded0fbeb1def4c82e352759089d80d40ab)

ticket: 9213
version_fixed: 1.22.3
---
 src/lib/gssapi/spnego/spnego_mech.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/lib/gssapi/spnego/spnego_mech.c b/src/lib/gssapi/spnego/spnego_mech.c
index 4a77836433..1dd0f17065 100644
--- a/src/lib/gssapi/spnego/spnego_mech.c
+++ b/src/lib/gssapi/spnego/spnego_mech.c
@@ -3517,6 +3517,8 @@ get_negTokenResp(OM_uint32 *minor_status, struct k5input *in,
 
 	if (k5_der_get_value(&seq, CONTEXT | 0x03, &field)) {
 		*mechListMIC = get_octet_string(&field);
+		if (*mechListMIC == GSS_C_NO_BUFFER)
+			return GSS_S_DEFECTIVE_TOKEN;
 
                 /* Handle Windows 2000 duplicate response token */
                 if (*responseToken &&
